File: easy/lettercase.lua
for line in io.lines(arg[1]) do
  local l = select(2, line:gsub("[a-z]", "."))
  local u = select(2, line:gsub("[A-Z]", "."))
  print(string.format("lowercase: %.2f uppercase: %.2f", 100*l/(l+u), 100*u/(l+u)))
end

File: easy/lettercase.hs
import System.Environment (getArgs)
import Data.Char (isLower, isUpper)
import Text.Printf (printf)

ratios  :: String -> String
ratios s = printf "lowercase: %.2f uppercase: %.2f" (100 * l / t) (100 * u / t)
         where l = fromIntegral (length [x | x <- s, isLower x]) :: Double
               u = fromIntegral (length [x | x <- s, isUpper x]) :: Double
               t = l + u

main :: IO ()
main = do
    [inpFile] <- getArgs
    input <- readFile inpFile
    putStr . unlines . map ratios $ lines input
